Jenkinsfile

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env groovy
+pipeline {
+  agent {
+    dockerfile true
+  }
+
+  environment {
+    JENKINS_VERSION = 'yes'
+    NODE_PATH = 'node_modules'
+  }
+
+  options {
+    timeout(time: 2, unit: 'HOURS')
```

Comment:
2hrs is incredibly long, especially for our code base. I would expect 1hr max.

Reply:
If there is a specific stage that we need to timeout we could only add to that stage.

Reply:
You're right that 2 hours is quite long and our builds should complete after 30 mins on average. I'll reduce the general timeout to 1 hour as it should be ok to flag a build that takes longer as a failure.

Reply:
Thanks!